docs/0.1/advanced/lorawan-devices/hackiot_philly.md

```diff
@@ -298,3 +298,88 @@ If you execute the request again, the number of records will increase as the dat
 
 Learn more on how to work collaboratively with other users by sharing records, check out our [Rules guide](../../../getting-started/rules-guide)
+
+<hr>
+
+## Decode the STM32 payloads
+
+You can now create a [Robot](../../../getting-started/robot-guide) to automatically decode all received STM32 payload in microshare. A Robot is a script driven entity that is triggered to run when a specific recType write operation occurs in the datalake.
+
+### Create Robot
+
+1. Go to [the Robot tab](https://app.microshare.io/composer#/robos) and click `CREATE`
+
+{% include image.html url="/assets/img/hackiot-create-a-robot.png" description="Create a Robot from the composer" %}
+
+We'll do the minimum to unlock all the Robot options for now.
+
+1. Give your Robot a name.
+2. Enter the exact Record Type you used in the Senet portal.
+3. Complete the creation by clicking the `CREATE` button.
+
+{% include image.html url="/assets/img/hackiot-create-a-robot-2.png" description="Minimal Robot configuration" %}
+
+You should be back in the Robot cards list and see your Robot should now be displayed.
+If you don't see your new Robot card listed:
+
+1. Open the option menu
+2. Increase the `Cards per Page` to 999 
+3. Click Apply
+
+The new Robot card should now be visible.
+
+{% include image.html url="/assets/img/hackiot-configure-robot.png" description="Increase Cards per Page" %}
+
+To edit an existing Robot, find your Robot in the list:
+
+1. Click on it 
+2. Click on the `pencil` icon at the top of the page
+
+{% include image.html url="/assets/img/hackiot-configure-robot-2.png" description="Open Robot edition mode" %}
+
+While in edit mode you can:
+1. Turn your Robot on and off
+2. Write the Robot script
+3. Test the script
+
+{% include image.html url="/assets/img/hackiot-configure-robot-3.png" description="Full Robot edition mode" %}
+
+Replace the code in your Robot script with:
+{% highlight js %}
+  var lib = require('./libs/helpers');
+  function main(text, auth) {
+      
+      var rec = lib.read(text, auth, []);
+      var m = rec.objs[0].data;
+      var recType = rec.objs[0].recType;
+      
+      var decodedLPP = lib.decodeCayenneLPP(m.pdu);    
+      var decodedLPPJSON = JSON.parse(decodedLPP);
+      
+      decodedLPPJSON.forEach(function(entry){
+          
+          print(entry);
+          print(JSON.stringify(entry));
+
+          lib.write(recType + '.decoded', entry, auth, []);
+
+      });
+  }
+{% endhighlight %}
+
+Be sure to change the `Write` recType (the first argument of the lib.write method) to decide which rectype the the decoded results will be saved to but **DO NOT set it to the same recType as the incoming Senet packages**, or you'll create a Robot triggering loop.
```

Comment:
This line isn't clear regarding recTypes.  Can you clarify this statement?  I'm not sure what the 2nd recType is.  I also suggest adding some comments to the JS Robot code so that it has some pointers or directions in the code itself.

@jgaye